tools/osv-scanner/main.go

```diff
@@ -119,18 +161,80 @@ func printResults(query osv.BatchedQuery, resp *osv.BatchedResponse) {
 }
 
 // TODO(ochang): Machine readable output format.
-// TODO(ochang): Ability to specify type of input.
 func main() {
-	flag.Usage = usage
-	flag.Parse()
-
 	var query osv.BatchedQuery
 
-	for _, arg := range flag.Args() {
-		if err := scan(&query, arg); err != nil {
-			log.Printf("scan failed: %v\n", err)
-			return
-		}
+	app := &cli.App{
+		Name:  "osv-scanner",
+		Usage: "scans various mediums for dependencies and matches it against the OSV database",
+		Flags: []cli.Flag{
+			&cli.StringSliceFlag{
+				Name:      "docker",
+				Aliases:   []string{"D"},
+				Usage:     "scan docker image with this name",
+				TakesFile: false,
+			},
+			&cli.StringSliceFlag{
+				Name:      "lockfile",
+				Aliases:   []string{"L"},
+				Usage:     "scan package lockfile on this path",
+				TakesFile: true,
+			},
+			&cli.StringSliceFlag{
+				Name:      "sbom",
+				Aliases:   []string{"S"},
+				Usage:     "scan sbom file on this path",
+				TakesFile: true,
+			},
+			&cli.StringSliceFlag{
+				Name:      "git",
+				Aliases:   []string{"G"},
+				Usage:     "scan for git repository in this directory",
+				TakesFile: true,
+			},
+		},
+		ArgsUsage: "[directory1 directory2...]",
```

Comment:
Does this refer to arbitrary directories being passed that don't come with a prefixing type flag? 

How is this handled right now?

Reply:
Yes, but it turns out I forgot to implement it. Added now in 030f8da